packages/devtools_app/lib/src/screens/inspector_v2/inspector_controls.dart

```diff
@@ -0,0 +1,132 @@
+            extensions.toggleSelectWidgetMode.extension,
+          ),
+          builder: (_, selectModeSupported, __) {
+            return ServiceExtensionButtonGroup(
+              extensions: [
+                selectModeSupported
+                    ? extensions.toggleSelectWidgetMode
+                    : extensions.toggleOnDeviceWidgetInspector,
+              ],
+              minScreenWidthForTextBeforeScaling:
+                  InspectorScreenBodyState.minScreenWidthForTextBeforeScaling,
+            );
+          },
+        ),
+        const SizedBox(width: defaultSpacing),
+        ShowImplementationWidgetsButton(controller: controller),
+        const Spacer(),
+        const SizedBox(width: defaultSpacing),
+        const InspectorServiceExtensionButtonGroup(),
+      ],
+    );
+  }
+}
+
+/// Group of service extension buttons for the inspector panel that control the
+/// overlays painted on the connected app.
+class InspectorServiceExtensionButtonGroup extends StatelessWidget {
+  const InspectorServiceExtensionButtonGroup({
+    super.key,
+  });
+
+  static const serviceExtensionButtonsIncludeTextWidth = 1200.0;
+
+  @override
+  Widget build(BuildContext context) {
+    return Row(
+      children: [
+        ServiceExtensionButtonGroup(
+          minScreenWidthForTextBeforeScaling:
+              serviceExtensionButtonsIncludeTextWidth,
+          extensions: [
+            extensions.slowAnimations,
+            extensions.debugPaint,
+            extensions.debugPaintBaselines,
+            extensions.repaintRainbow,
+            extensions.invertOversizedImages,
+          ],
+        ),
+        const SizedBox(width: defaultSpacing),
+        SettingsOutlinedButton(
+          gaScreen: gac.inspector,
+          gaSelection: gac.inspectorSettings,
+          tooltip: 'Flutter Inspector Settings',
+          onPressed: () {
+            unawaited(
+              showDialog(
+                context: context,
+                builder: (context) => const FlutterInspectorSettingsDialog(),
+              ),
+            );
+          },
+        ),
+      ],
+    );
+  }
+}
+
+/// Toggle button that allows showing/hiding the implementation widgets in the
+/// widget tree.
+class ShowImplementationWidgetsButton extends StatelessWidget {
+  const ShowImplementationWidgetsButton({
+    super.key,
+    required this.controller,
+  });
+
+  final InspectorController controller;
+
+  @override
+  Widget build(BuildContext context) {
+    return ValueListenableBuilder(
+      valueListenable: controller.implementationWidgetsHidden,
+      builder: (context, isHidden, _) {
+        return DevToolsToggleButton(
+          isSelected: !isHidden,
+          message:
+              'Show widgets created by the Flutter framework or other packages.',
+          label: 'Show Implementation Widgets',
+          onPressed: () async {
+            await controller.toggleImplementationWidgetsVisibility();
```

Comment:
Nit: I _think_ this can just be `onPressed: controller.toggleImplementationWidgetsVisibility`. If `onPressed` takes `Future Function()`, it should await the returned `Future`. If `onPressed` doesn't await it, this await doesn't really accomplish anything.

Reply:
Yes, that works. Thanks!